crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -770,6 +770,95 @@ async fn test_sync_message_is_indexed() {
     assert_eq!(response[0], event_id, "event id doesn't match: {response:?}");
 }
 
+#[cfg(feature = "experimental-search")]
+#[async_test]
+async fn test_search_index_edit_ordering() {
+    let room_id = room_id!("!room_id:localhost");
+    let dummy_id = event_id!("$dummy");
+    let edit1_id = event_id!("$edit1");
+    let edit2_id = event_id!("$edit2");
+    let edit3_id = event_id!("$edit3");
+    let original_id = event_id!("$original");
+
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().build().await;
+
+    let event_cache = client.event_cache();
+    event_cache.subscribe().unwrap();
+
+    let room = server.sync_joined_room(&client, room_id).await;
+
+    let f = EventFactory::new().room(room_id).sender(user_id!("@user_id:localhost"));
+
+    // Indexable dummy message required because RoomIndex is initialised lazily.
+    let dummy = f.text_msg("dummy").event_id(dummy_id).into_raw();
+
+    let original = f.text_msg("This is a message").event_id(original_id).into_raw();
+
+    let edit1 = f
+        .text_msg("* A new message")
+        .edit(original_id, RoomMessageEventContentWithoutRelation::text_plain("A new message"))
+        .event_id(edit1_id)
+        .into_raw();
+
+    let edit2 = f
+        .text_msg("* An even newer message")
+        .edit(
+            original_id,
+            RoomMessageEventContentWithoutRelation::text_plain("An even newer message"),
+        )
+        .event_id(edit2_id)
+        .into_raw();
+
+    let edit3 = f
+        .text_msg("* The newest message")
+        .edit(original_id, RoomMessageEventContentWithoutRelation::text_plain("The newest message"))
+        .event_id(edit3_id)
+        .into_raw();
+
+    server
+        .sync_room(
+            &client,
+            JoinedRoomBuilder::new(room_id)
+                .add_timeline_event(dummy.clone())
+                .add_timeline_event(edit1.clone())
+                .add_timeline_event(edit2.clone()),
+        )
+        .await;
+
+    let results = room.search("message", 3).await.unwrap();
+
+    assert_eq!(results.len(), 0, "Search should return 0 results, got {results:?}");
+
+    // Adding the original after some pending edits should add the latest edit
+    // instead of the original.
+    server
+        .mock_sync()
+        .ok_and_run(&client, |builder| {
+            builder.add_joined_room(JoinedRoomBuilder::new(room_id).add_timeline_event(original));
+        })
```

Comment:
nit: here and below, you can use `.sync_room(JoinedRoomBuilder::new(room_id).add_timeline_event(original)))` too.